odl/tomo/backends/astra_setup.py

```diff
@@ -0,0 +1,581 @@
+    else:
+        raise ValueError('{}-dimensional volume geometries not supported '
+                         'by ASTRA.'.format(discr_reco.ndim))
+    return vol_geom
+
+
+def astra_geom_to_vec(geometry):
+    """Create vectors for ASTRA projection geometries from `odl.tomo.geometry`.
+
+     The 3D vectors are used to create an ASTRA projection geometry for
+     cone beam geometries ('conve_vec') with helical acquisition curves.
+
+     Additionally, supports creation of 2D and 3D vectors for 'flat_vec' or
+     'parallel3d_vec', respectively.
+
+    Output vectors:
+
+    2d geometry 'fanflat_vec':
+    Each row of vectors corresponds to a single projection, and consists of:
+        ( srcX, srcY, dX, dY, uX, uY )
+        src : the ray source
+        d : the center of the detector
+        u : the vector between the centers of detector pixels 0 and 1
+
+    3d geometry 'parallel3d_vec':
+    Each row of vectors corresponds to a single projection, and consists of:
+        ( rayX, rayY, rayZ, dX, dY, dZ, uX, uY, uZ, vX, vY, vZ )
+        ray : the ray direction
+        d   : the center of the detector
+        u   : the vector from detector pixel (0,0) to (0,1)
+        v   : the vector from detector pixel (0,0) to (1,0)
+
+    3d geometry 'cone_vec':
+    Each row of vectors corresponds to a single projection, and consists of:
+        ( srcX, srcY, srcZ, dX, dY, dZ, uX, uY, uZ, vX, vY, vZ ):
+        src : the ray source
+        d   : the center of the detector
+        u   : the vector from detector pixel (0,0) to (0,1)
+        v   : the vector from detector pixel (0,0) to (1,0)
+
+    Parameters
+    ----------
+    geometry : `Geometry`
+        The `odl.tomo.geometry` instance used to create the ASTRA geometry
+
+    Returns
+    -------
+    vectors : `numpy.ndarray`
+        Numpy array of shape ``(number of angles, 12)``
+    """
+
+    angles = geometry.angle_grid
+    num_angles = geometry.angle_grid.ntotal
+
+    if isinstance(geometry, (CircularConeFlatGeometry,
+                             HelicalConeFlatGeometry)):
+        vectors = np.zeros((num_angles, 12))
+        det_pix_width = geometry.det_grid.stride[0]
+        det_pix_height = geometry.det_grid.stride[1]
+
+        for and_ind in range(num_angles):
+            angle = angles[and_ind][0]
+
+            # source position
+            # TODO: check geometry class for consistency since 'src_position'
+            # and 'det_refpoint' were adopted to use ASTRA cone_vec convention
+            vectors[and_ind, 0:3] = geometry.src_position(angle)
+
+            # center of detector
+            vectors[and_ind, 3:6] = geometry.det_refpoint(angle)
+
+            # vector from detector pixel (0,0) to (0,1)
+            # TODO: use geometry class method 'det_rotation' method instead
+            vectors[and_ind, 6] = cos(angle) * det_pix_width
+            vectors[and_ind, 7] = sin(angle) * det_pix_width
```

Comment:
You are indeed correct. In that case, this does seem to ignore the `axis` option of `ConeBeamGeometry`.